pkg/webhook/clusterset/validatingWebhook.go

```diff
@@ -176,34 +153,79 @@ func (a *AdmissionHandler) validateCreateRequest(request *v1.AdmissionRequest) *
 
 // validateUpdateRequest validates:
 // 1. allow if the managedClusterSet label is not changed.
-// 2. the user must have managedClusterSet/join <clusterSet-A/clusterSet-B> permission if update resource to change
-// the managedClusterSet label from clusterSet-A to clusterSet-B.
-// 3. the user must have managedClusterSet/join <all> permission if update resource to add/remove the managedClusterSet label.
+// 2. For clusterpool, do not allow to update managedClusterSet label
+// 3. For managedClusters, when their is no clusterdeployment (import a existing cluster), or the clusterdeployment is not
+//    refer to a clusterpool(directlly create a cluster), we need to check if he has permission to update the managedClusterSet label.
+//   3.1. the user must have managedClusterSet/join <clusterSet-A/clusterSet-B> permission if update resource to change
+//         the managedClusterSet label from clusterSet-A to clusterSet-B.
+//   3.2. the user must have managedClusterSet/join <all> permission if update resource to add/remove the managedClusterSet label.
+// 4. if the managedcluster claimed from a clusterpool, we will always sync the managedcluster clusterset by clusterdeployment.
```

Comment:
this is confusing...would rephrase the comment here?

Reply:
done